keras/layers/cudnn_recurrent.py

```diff
@@ -47,7 +47,7 @@ def __init__(self,
 
     def _canonical_to_params(self, weights, biases):
         import tensorflow as tf
-        weights = [tf.reshape(x, (-1,)) for x in weights]
+        weights = [tf.reshape(tf.transpose(x), (-1,)) for x in weights]
```

Comment:
I think this would break existing checkpoints. Have you tried?

Reply:
Yes it's a breaking change. If it's important not to break existing checkpoints, I can reshape the kernel during weight loading. For example, adding these lines in `preprocess_weights_for_loading()` will also work:
```python
# reshape the kernels
kernels = np.split(weights[0], 4, axis=1)
kernels = [kernel.reshape(-1).reshape(kernel.shape, order='F') for kernel in kernels]
weights[0] = np.concatenate(kernels, axis=1)

# reshape the recurrent kernels
recurrent_kernels = np.split(weights[1], 4, axis=1)
recurrent_kernels = [kernel.T for kernel in recurrent_kernels]
weights[1] = np.concatenate(recurrent_kernels, axis=1)
```
But if we fix it like this, the kernels in `CuDNNLSTM` and `CuDNNGRU` are still in wrong order. I think it would be cleaner to fix it in `_canonical_to_params()`.